core/io/file_access_encrypted.cpp

```diff
@@ -210,10 +210,16 @@ bool FileAccessEncrypted::eof_reached() const {
 }
 
 uint64_t FileAccessEncrypted::get_buffer(uint8_t *p_dst, uint64_t p_length) const {
-	ERR_FAIL_COND_V(!p_dst && p_length > 0, -1);
 	ERR_FAIL_COND_V_MSG(writing, -1, "File has not been opened in read mode.");
 
+	if (!p_length) {
+		return 0;
+	}
+
+	ERR_FAIL_COND_V(!p_dst, -1);
```

Comment:
```suggestion
	ERR_FAIL_NULl_V(p_dst, -1);
```
Now that it's not a compound expression

Reply:
Ah, you're right, I'll update. I can't believe this didn't make tests fail. 😅

Reply:
It is valid code, we (unfortunately) can't catch pointer bool checks (it fails for `Ref`)